<a href="https://colab.research.google.com/github/Ganesh-Esc/Basic-Exploratory-Data-Analysis-/blob/main/Copy_of_Data_Preprocessing1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>



Identifying Null Values



In [ ]:
! pip install pandas

In [ ]:
import pandas as pd
df=pd.read_csv('googleplaystore_v2.csv')

In [ ]:
df.head()

App        Category  Rating  \
0     Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN     4.1   
1                                Coloring book moana  ART_AND_DESIGN     3.9   
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ART_AND_DESIGN     4.7   
3                              Sketch - Draw & Paint  ART_AND_DESIGN     4.5   
4              Pixel Draw - Number Art Coloring Book  ART_AND_DESIGN     4.3   

  Reviews     Size     Installs  Type Price Content Rating  \
0     159  19000.0      10,000+  Free     0       Everyone   
1     967  14000.0     500,000+  Free     0       Everyone   
2   87510   8700.0   5,000,000+  Free     0       Everyone   
3  215644  25000.0  50,000,000+  Free     0           Teen   
4     967   2800.0     100,000+  Free     0       Everyone   

                      Genres      Last Updated         Current Ver  \
0               Art & Design   January 7, 2018               1.0.0   
1  Art & Design;Pretend Play  January 15, 2018               2.0.0   
2               Art & Design    August 1, 2018               1.2.4   
3               Art & Design      June 8, 2018  Varies with device   
4    Art & Design;Creativity     June 20, 2018                 1.1   

    Android Ver  
0  4.0.3 and up  
1  4.0.3 and up  
2  4.0.3 and up  
3    4.2 and up  
4    4.4 and up

In [ ]:
print(df.isnull())

         App  Category  Rating  Reviews   Size  Installs   Type  Price  \
0      False     False   False    False  False     False  False  False   
1      False     False   False    False  False     False  False  False   
2      False     False   False    False  False     False  False  False   
3      False     False   False    False  False     False  False  False   
4      False     False   False    False  False     False  False  False   
...      ...       ...     ...      ...    ...       ...    ...    ...   
10836  False     False   False    False  False     False  False  False   
10837  False     False   False    False  False     False  False  False   
10838  False     False    True    False  False     False  False  False   
10839  False     False   False    False  False     False  False  False   
10840  False     False   False    False  False     False  False  False   

       Content Rating  Genres  Last Updated  Current Ver  Android Ver  
0               False   False         F

In [ ]:
print(df.isnull().sum())

App                  0
Category             0
Rating            1474
Reviews              0
Size                 0
Installs             0
Type                 1
Price                0
Content Rating       1
Genres               0
Last Updated         0
Current Ver          8
Android Ver          3
dtype: int64


Handling Null Values

In [ ]:
df1=df.dropna()
print(df1.isnull().sum())

App               0
Category          0
Rating            0
Reviews           0
Size              0
Installs          0
Type              0
Price             0
Content Rating    0
Genres            0
Last Updated      0
Current Ver       0
Android Ver       0
dtype: int64


SO we have removed all the rows which has NaN or null values That's how we removed (1474+1+1+8+3) rows from the actual dataset

Numerical Data Analysis

In this data set we have rating and Reviews as numbers

Average Rating

In [ ]:
# Calculate the average rating using the .mean() method, which handles NaN values
average_rating = df['Rating'].mean()
print(f"The average rating is: {average_rating:.2f}")

The average rating is: 4.19


In [ ]:
count=0
for i in df['Rating']:
  if i==5.0:
    count+=1
print("The count of apps which has a rating of 5 is:",count)

The count of apps which has a rating of 5 is: 274


In [ ]:
# Convert the 'Reviews' column to numeric, coercing errors to NaN
df['Reviews'] = pd.to_numeric(df['Reviews'], errors='coerce')

# Calculate the average number of reviews
average_reviews = df['Reviews'].mean()
print("The average reviews is:",average_reviews)

The average reviews is: 444152.89603321033


This is because of type conversion string-> int  reviews are stored as string data type

Data Analysis Categorical

In [ ]:
df['Category'].unique()

array(['ART_AND_DESIGN', 'AUTO_AND_VEHICLES', 'BEAUTY',
       'BOOKS_AND_REFERENCE', 'BUSINESS', 'COMICS', 'COMMUNICATION',
       'DATING', 'EDUCATION', 'ENTERTAINMENT', 'EVENTS', 'FINANCE',
       'FOOD_AND_DRINK', 'HEALTH_AND_FITNESS', 'HOUSE_AND_HOME',
       'LIBRARIES_AND_DEMO', 'LIFESTYLE', 'GAME', 'FAMILY', 'MEDICAL',
       'SOCIAL', 'SHOPPING', 'PHOTOGRAPHY', 'SPORTS', 'TRAVEL_AND_LOCAL',
       'TOOLS', 'PERSONALIZATION', 'PRODUCTIVITY', 'PARENTING', 'WEATHER',
       'VIDEO_PLAYERS', 'NEWS_AND_MAGAZINES', 'MAPS_AND_NAVIGATION',
       '1.9'], dtype=object)

In [ ]:
n=df['Category'].nunique()
print("The number of unique categories is:",n)

The number of unique categories is: 34


In [ ]:
c=0
for i in df['Category']:
  if i == 'ART_AND_DESIGN':
    c+=1
print(c)

65


There are total of 65 apps in ART_AND_DESIGN category

In [ ]:
m=df['Type'].nunique()
m

3

In [ ]:
df['Type'].unique()

array(['Free', 'Paid', nan, '0'], dtype=object)

In [ ]:
f,p=0,0
for i in df['Type']:
  if i=='Free':
    f+=1
  if i=='Paid':
    p+=1
print("The number of free apps is:",f)
print("The number of paid apps is:",p)

The number of free apps is: 10039
The number of paid apps is: 800


Data Analysis Automatic Categotical

Counting Apps in each category

In [ ]:
categories={}
for name in df['Category']:
  ct=0
  for i in df['Category']:
    if i==name:
      ct+=1
  categories[name]=ct
print(categories)

{'ART_AND_DESIGN': 65, 'AUTO_AND_VEHICLES': 85, 'BEAUTY': 53, 'BOOKS_AND_REFERENCE': 231, 'BUSINESS': 460, 'COMICS': 60, 'COMMUNICATION': 387, 'DATING': 234, 'EDUCATION': 156, 'ENTERTAINMENT': 149, 'EVENTS': 64, 'FINANCE': 366, 'FOOD_AND_DRINK': 127, 'HEALTH_AND_FITNESS': 341, 'HOUSE_AND_HOME': 88, 'LIBRARIES_AND_DEMO': 85, 'LIFESTYLE': 382, 'GAME': 1144, 'FAMILY': 1972, 'MEDICAL': 463, 'SOCIAL': 295, 'SHOPPING': 260, 'PHOTOGRAPHY': 335, 'SPORTS': 384, 'TRAVEL_AND_LOCAL': 258, 'TOOLS': 843, 'PERSONALIZATION': 392, 'PRODUCTIVITY': 424, 'PARENTING': 60, 'WEATHER': 82, 'VIDEO_PLAYERS': 175, 'NEWS_AND_MAGAZINES': 283, 'MAPS_AND_NAVIGATION': 137, '1.9': 1}


In [ ]:
categories={}
for name in df['Category']:
  if name in categories:
    categories[name]+=1
  else:
    categories[name]=1
print(categories)


{'ART_AND_DESIGN': 65, 'AUTO_AND_VEHICLES': 85, 'BEAUTY': 53, 'BOOKS_AND_REFERENCE': 231, 'BUSINESS': 460, 'COMICS': 60, 'COMMUNICATION': 387, 'DATING': 234, 'EDUCATION': 156, 'ENTERTAINMENT': 149, 'EVENTS': 64, 'FINANCE': 366, 'FOOD_AND_DRINK': 127, 'HEALTH_AND_FITNESS': 341, 'HOUSE_AND_HOME': 88, 'LIBRARIES_AND_DEMO': 85, 'LIFESTYLE': 382, 'GAME': 1144, 'FAMILY': 1972, 'MEDICAL': 463, 'SOCIAL': 295, 'SHOPPING': 260, 'PHOTOGRAPHY': 335, 'SPORTS': 384, 'TRAVEL_AND_LOCAL': 258, 'TOOLS': 843, 'PERSONALIZATION': 392, 'PRODUCTIVITY': 424, 'PARENTING': 60, 'WEATHER': 82, 'VIDEO_PLAYERS': 175, 'NEWS_AND_MAGAZINES': 283, 'MAPS_AND_NAVIGATION': 137, '1.9': 1}


Content Rating Analysis

In [ ]:
ratings={}
for name in df['Content Rating']:
  if name in ratings:
    ratings[name]+=1
  else:
    ratings[name]=1
print(ratings)

{'Everyone': 8714, 'Teen': 1208, 'Everyone 10+': 414, 'Mature 17+': 499, 'Adults only 18+': 3, 'Unrated': 2, nan: 1}


In [ ]:
df['Content Rating'].describe()

count        10840
unique           6
top       Everyone
freq          8714
Name: Content Rating, dtype: object